platform/swift/source/src/bridge.rs

```diff
@@ -567,6 +581,126 @@ extern "C" fn capture_process_issue_reports(
   );
 }
 
+#[no_mangle]
+extern "C" fn capture_persist_javascript_error_report(
+  logger_id: LoggerId<'_>,
+  error_name: *const c_char,
+  error_message: *const c_char,
+  stack_trace: *const c_char,
+  is_fatal: bool,
+  engine: *const c_char,
+  debugger_id: *const c_char,
+  timestamp_millis: i64,
+  destination_path: *const c_char,
+) {
+  with_handle_unexpected(
+    || -> anyhow::Result<()> {
+      let error_name_str = unsafe { CStr::from_ptr(error_name) }.to_str()?;
+      let error_message_str = unsafe { CStr::from_ptr(error_message) }.to_str()?;
+      let stack_trace_str = unsafe { CStr::from_ptr(stack_trace) }.to_str()?;
+      let engine_str = unsafe { CStr::from_ptr(engine) }.to_str()?;
+      let destination_str = unsafe { CStr::from_ptr(destination_path) }.to_str()?;
+
+      let debugger_id_opt = if debugger_id.is_null() {
+        None
+      } else {
+        let id_str = unsafe { CStr::from_ptr(debugger_id) }.to_str()?;
+        if id_str.is_empty() {
+          None
+        } else {
+          Some(id_str)
+        }
+      };
+
+      let timestamp_seconds = u64::try_from(timestamp_millis / 1_000).unwrap_or_default();
+      let timestamp_nanos = u32::try_from((timestamp_millis % 1_000) * 1_000).unwrap_or_default();
+
+      let metadata_provider = logger_id.metadata_provider();
+      let (ootb_fields, _custom_fields) = metadata_provider.fields()?;
```

Comment:
this call is a bit scary since it ends up calling back into all the user provided FieldProviders in addition the default ones. Using global_state::Reader might be safer until we can redo the interfaces to not have to rely on field providers for all this